workspaces/arborist/lib/node.js

```diff
@@ -1344,9 +1376,101 @@ class Node {
     this.edgesOut.set(edge.name, edge)
   }
 
-  addEdgeIn (edge) {
+  recalculateOutEdgesOverrides () {
+    // For each edge out propogate the new overrides through.
+    for (const edge of this.edgesOut.values()) {
+      edge.reload(true)
+      if (edge.to) {
+        edge.to.updateOverridesEdgeInAdded(edge.overrides)
+      }
+    }
+  }
+
+  static findSpecificOverrideSet (first, second) {
```

Comment:
After looking at how the code interacts with this and to avoid cyclical module use we can move `findSpecificOverrideSet` to its own file like `lib/dep-valid.js`, so `lib/find-specific-override-set.js` or however you name it. This way it avoids the `.constructor` stuff and allows both `edge.js` and `node.js` to require it.

Reply:
Better to just move it `override-set.js`, since it fundamentally compares two override sets as is.

Reply:
If the `node` module is already importing the `edge` module then it seems fine to me too.